source/grid/grid_in.cc

```diff
@@ -3252,6 +3252,79 @@ GridIn<dim, spacedim>::read_partitioned_msh(const std::string &file_prefix,
           coords[3 * i + d];
     }
 
+  // --- Collect physical groups for boundary and material info ---
+  std::map<std::pair<int, int>, types::boundary_id> entity_to_boundary;
+  std::map<std::pair<int, int>, types::material_id> entity_to_material;
+
+
+  std::vector<std::pair<int, int>> physical_groups;
+  gmsh::model::getPhysicalGroups(physical_groups);
+
+  for (const auto &pg : physical_groups)
+    {
+      int physical_dim = pg.first;
+      int physical_tag = pg.second;
```

Comment:
Unimportant for this patch, but for future cases: This can be written more concisely with C++17 as
```suggestion
  for (const auto &[physical_dim, physical_tag]: physical_groups)
    {
```